docs/content/querying/filters.md

```diff
@@ -58,18 +58,28 @@ The filter specified at field can be any other filter defined on this page.
 
 ### JavaScript filter
 
-The JavaScript filter matches a dimension against the specified JavaScript function predicate. The filter matches values for which the function returns true.
+The JavaScript filter matches dimensions against the specified JavaScript function predicate. The filter matches values for which the function returns true.
 
-The function takes a single argument, the dimension value, and returns either true or false.
+The function takes same number of arguments as the dimension values, and returns either true or false.
 
 ```json
 {
   "type" : "javascript",
-  "dimension" : <dimension_string>,
-  "function" : "function(value) { <...> }"
+  "dimension" : <a dimension_string>,
```

Comment:
+1 for removing `dimension` from docs. also, let us explicitly call it deprecated in 0.9.0 release notes so that we can remove it in later release from the code.